Assets/Modules/Obstacles/Scripts/ObstacleHookController.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ObstacleHookController : MonoBehaviour
+{
+    [SerializeField] private ObstacleEntity obstacleEntity;
+    [SerializeField] float reactivateHookAfter = 5f;
+
+    private SpriteRenderer sr;
+    private float timerForHookReactivation = 0;
+
+    private void Awake() 
+    {
+        sr = GetComponent<SpriteRenderer>();
+    }
+    
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (!obstacleEntity.hooknessActivated)
+        {
+            timerForHookReactivation += Time.deltaTime;
+            if (timerForHookReactivation >= reactivateHookAfter)
+            {
+                ReactivateHook();
+            }
+        }
+    }
+
+    public void DeactivateHook()
+    {
+        Color tmp = sr.color;
+        tmp.a = 0f;
+        sr.color = tmp;
+        obstacleEntity.hooknessActivated = false;
+        Destroy(GetComponent<CircleCollider2D>());
+    }
+
+    public void ReactivateHook()
+    {
+        timerForHookReactivation = 0;
+        Color tmp = sr.color;
+        tmp.a = 255f;
+        sr.color = tmp;
+        obstacleEntity.hooknessActivated = true;
+        gameObject.AddComponent<CircleCollider2D>();
+        CircleCollider2D myCollider = gameObject.GetComponent<CircleCollider2D>();
```

Comment:
I think this one I can't SerializeField because it's destroyed at some point, am I wrong?
So as far as I understand, serialized fields, I can assign them in the Inspector, but if I destroy that component, when I recreate it in the future (which is what I'm doing when I "ReactivateHook()", it won't be assigned in the serialized field anymore right?

Reply:
Yes exactly! They refer to an instance of that class that you define and if that instance gets destroyed, you need to re-assign it. For now, let's keep it like this. We can think about how to do it properly in future :)